crates/shared/src/sources/balancer_v2/pool_cache.rs

```diff
@@ -71,50 +72,26 @@ impl CacheFetching<H256, WeightedPool> for PoolReserveFetcher {
         pool_ids: HashSet<H256>,
         at_block: Block,
     ) -> Result<Vec<WeightedPool>> {
-        let mut batch = CallBatch::new(self.web3.transport());
+        let mut batch = CallBatch::new(self.web3.transport().clone());
```

Comment:
Just because we made the type alias be `type Web3CallBatch = CallBatch<Web3Transport>` instead of `CallBatch<&Web3Transport>`. The choice was just to simplify lifetimes and `mockall` interoperability (it has the same issue as `BoxFuture<'a, ...>` not being able to specify the `'a`). Since transport are generally `Arc<...>` wrapped anyway, this should not have a significant impact.